src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -3462,24 +3485,31 @@ internal static bool TryGetStaticEntitySet(this IEdmPathExpression pathExpressio
         /// </summary>
         /// <param name="type">Reference to the calling object.</param>
         /// <param name="model">The model to be used.</param>
+        /// <param name="alternateKeyTerms">
+        /// The <see cref="IEdmTerm"/>s that are used within <paramref name="model"/> to represent alternate key annotations; assumed to not be <see langword="null"/>
+        /// </param>
         /// <returns>Alternate Keys of this type.</returns>
-        private static IEnumerable<IDictionary<string, IEdmProperty>> GetDeclaredAlternateKeysForType(IEdmEntityType type, IEdmModel model)
+        /// <exception cref="ArgumentException">Thrown if <paramref name="alternateKeyTerms"/> contains any <see langword="null"/> values</exception>
+        private static IEnumerable<IDictionary<string, IEdmProperty>> GetDeclaredAlternateKeysForType(
+            IEdmEntityType type,
+            IEdmModel model, 
+            IEnumerable<IEdmTerm> alternateKeyTerms)
         {
-            IEdmVocabularyAnnotation annotationValue = model.FindVocabularyAnnotations<IEdmVocabularyAnnotation>(type, AlternateKeysVocabularyModel.AlternateKeysTerm).FirstOrDefault();
-            IEdmVocabularyAnnotation coreAnnotationValue = model.FindVocabularyAnnotations<IEdmVocabularyAnnotation>(type, CoreVocabularyModel.AlternateKeysTerm).FirstOrDefault();
-
-            if (annotationValue != null || coreAnnotationValue != null)
+            var any = false;
+            var declaredAlternateKeys = new List<IDictionary<string, IEdmProperty>>();
+            foreach (var alternateKeyTerm in alternateKeyTerms)
             {
-                List<IDictionary<string, IEdmProperty>> declaredAlternateKeys = new List<IDictionary<string, IEdmProperty>>();
+                if (alternateKeyTerm == null)
+                {
+                    throw new ArgumentException(Strings.NullTermForAlternateKey(nameof(alternateKeyTerms)));
+                }
 
-                Action<IEdmVocabularyAnnotation> retrieveAnnotationAction = ann =>
+                var annotationValue = model.FindVocabularyAnnotations<IEdmVocabularyAnnotation>(type, alternateKeyTerm).FirstOrDefault();
+                if (annotationValue != null)
                 {
-                    if (ann == null)
-                    {
-                        return;
-                    }
+                    any = true;

```

Comment:
>any = true; [](http://example.com/codeflow?start=20&length=11)

should we wait to set this until we have a value (i.e., versus the collectionexpression being empty)?